File: epi_judge_java/epi/StringIntegerInterconversion.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;
import epi.test_framework.TestFailure;

import java.util.HashMap;
import java.util.Map;

public class StringIntegerInterconversion {
    public static String intToString(int x) {
        if (x == 0) {
            return "0";
        }

        boolean negative = false;
        if (x < 0) {
            negative = true;
        }

        StringBuilder sBuilder = new StringBuilder();

        int carry = x;
        while (carry != 0) {
            sBuilder.append(Math.abs(carry % 10));
            carry /= 10;
        }

        if (negative) {
            sBuilder.append("-");
        }

        return sBuilder.reverse().toString();
    }

    public static int stringToInt(String s) {
        boolean negative = false;
        if (s.charAt(0) == '-') {
            negative = true;
        }

        Map<Character, Integer> digits = new HashMap<>();
        digits.put('0', 0);
        digits.put('1', 1);
        digits.put('2', 2);
        digits.put('3', 3);
        digits.put('4', 4);
        digits.put('5', 5);
        digits.put('6', 6);
        digits.put('7', 7);
        digits.put('8', 8);
        digits.put('9', 9);

        int result = 0;
        for (int i = 0; i < s.length(); i++) {
            if (digits.containsKey(s.charAt(i))) {
                result *= 10;
                result += digits.get(s.charAt(i));
            }
        }

        if (negative) {
            result *= -1;
        }
        return result;
    }

    @EpiTest(testDataFile = "string_integer_interconversion.tsv")
    public static void wrapper(int x, String s) throws TestFailure {
        if (!intToString(x).equals(s)) {
            throw new TestFailure("Int to string conversion failed");
        }
        if (stringToInt(s) != x) {
            throw new TestFailure("String to int conversion failed");
        }
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "StringIntegerInterconversion.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_python/string_integer_interconversion.py
from test_framework import generic_test
from test_framework.test_failure import TestFailure

CH_TO_DIGIT = {str(i): i for i in range(10)}
DIGIT_TO_CH = {i: str(i) for i in range(10)}

def int_to_string(x) -> str:
    chars = list()
    sign = 1

    if x == 0:
        chars.append(DIGIT_TO_CH[0])
    if x < 0:
        sign = -1
        x *= -1

    while x != 0:
        chars.append(DIGIT_TO_CH[x % 10])
        x //= 10

    if sign == -1:
        chars.append('-')
    return ''.join(reversed(chars))


def string_to_int(s) -> int:
    num = 0
    sign = 1
    for ch in s:
        if ch == '-':
            sign = -1
        else:
            num = num * 10 + CH_TO_DIGIT[ch]

    return num * sign


def wrapper(x, s):
    if int_to_string(x) != s:
        raise TestFailure("Int to string conversion failed")
    if string_to_int(s) != x:
        raise TestFailure("String to int conversion failed")


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("string_integer_interconversion.py",
                                       'string_integer_interconversion.tsv',
                                       wrapper))
